pkg/nmstatectl/nmstatectl.go

```diff
@@ -57,7 +57,12 @@ func Show(arguments ...string) (string, error) {
 }
 
 func Set(desiredState nmstatev1alpha1.State, timeout time.Duration) (string, error) {
-	return nmstatectlWithInput([]string{"set", "--no-commit", "--timeout", strconv.Itoa(int(timeout.Seconds()))}, string(desiredState.Raw))
+	var setDoneCh = make(chan struct{})
+	go setUnavailableUp(setDoneCh)
+	defer close(setDoneCh)
```

Comment:
a go routine waits for all its children ? (asking because the defer here that will close the channel)

Reply:
SetUnavailableUp will be running until we close the stopChannel which is only after the `set` is done. What is the scenario that would result in a race you think of?